File: typescript/src/utils/converse.ts
import {
	BedrockRuntimeClient,
	ConverseCommand,
} from "@aws-sdk/client-bedrock-runtime";

const bedrock = new BedrockRuntimeClient({});

export async function converse(
	modelId: string,
	prompt: string,
): Promise<string> {
	const response = await bedrock.send(
		new ConverseCommand({
			modelId,
			messages: [{ role: "user", content: [{ text: prompt }] }],
		}),
	);
	return response.output?.message?.content?.[0].text ?? "";
}

File: python/src/utils/converse.py
import boto3

_bedrock = None


def _get_bedrock():
    global _bedrock
    if _bedrock is None:
        _bedrock = boto3.client("bedrock-runtime")
    return _bedrock


def converse(model_id: str, prompt: str) -> str:
    response = _get_bedrock().converse(
        modelId=model_id,
        messages=[{"role": "user", "content": [{"text": prompt}]}],
    )
    return (
        response.get("output", {})
        .get("message", {})
        .get("content", [{}])[0]
        .get("text", "")
    )
